Automattic/jetpack
# Masterbar: Remove feature inclusion on Jetpack self-hosted sites

Fixes https://github.com/Automattic/vulcan/issues/446

## Proposed changes:

* This PR removes the Masterbar functionality from the Jetpack plugin, ensuring the Masterbar features aren't loaded on Jetpack self-hosted sites. However the feature should continue to load on WoA and Simple sites.
* The module itself technically still exists (the file hasn't been removed yet), so the corresponding card on the Jetpack > Settings > Writing page has been removed, and the module activation / deactivation option from the modules page at `/wp-admin/admin.php?page=jetpack_modules`
  * The removal of the module itself is set for the following release: see pfwV0U-3U-p2#comment-327
* On self-hosted sites, a notice should appear regarding the removal.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/446

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* The testing instructions on this PR apply here, but noting that self-hosted sites should not have any expected Masterbar related functionality: https://github.com/Automattic/jetpack/pull/37676

On self-hosted sites:
* On the Jetpack modules page, there should be no listing for the WordPress.com Toolbar: `/wp-admin/admin.php?page=jetpack_modules`
* On the Jetpack > Settings > Writing page, there should be no card for the WordPress.com Toolbar: `/wp-admin/admin.php?page=jetpack#/writing`
* On Jetpack dashboard pages, you should see a notice mentioning the removal, assuming you had the feature enabled before applying the PR (this can be done via `/wp-admin/admin.php?page=jetpack_modules` before applying the PR).
* The notice should also be visible on other `wp-admin` pages, as well as in My Jetpack. (The support doc itself is yet to be updated).
* Clicking dismiss should hide the banner.
* There should be no Masterbar related functionality (see https://github.com/Automattic/jetpack/pull/37676 for how to test).

On WoA sites:
* The testing instructions on this PR apply here: https://github.com/Automattic/jetpack/pull/37676
* There should be no notice on Jetpack dashboard pages, given the same constraints as on a self-hosted site. Functionality should be as expected.
